internal/controller/util/nslock.go

```diff
@@ -0,0 +1,45 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"sync"
+)
+
+// NamespaceLock implements atomic operation for namespace. It will have the namespace
+// having multiple vrgs in which VRGs are being processed.
+type NamespaceLock struct {
+	namespace string
+	mux       sync.Mutex
+}
+
+// NewNamespaceLock returns new NamespaceLock
+func NewNamespaceLock() *NamespaceLock {
```

Comment:
@raghavendra-talur : Pass ns string and use for struct init.

Reply:
Just gave a thought on what we discussed. volumereplicationgroup_controller -- SetupWithManager (locks are initialized) r.locks = rmnutil.NewNamespaceLock()  ns cannot be passed here as this call is well before Reconcile is invoked. So instead of string with ns, set of namespaces(locked) ones would be better.

Reply:
In that case, it could be
```
type NamespaceLock struct {
	mux       map[string]sync.Mutex
}
```